src/coreclr/src/jit/lowerxarch.cpp

```diff
@@ -1081,6 +1092,936 @@ void Lowering::LowerHWIntrinsic(GenTreeHWIntrinsic* node)
+
+                    node->gtOp1 = op1;
+                    node->gtOp2 = nullptr;
+
+                    node->gtHWIntrinsicId = NI_SSE3_MoveAndDuplicate;
+                    break;
+                }
+
+                assert(comp->compIsaSupportedDebugOnly(InstructionSet_SSE2));
+
+                //      /--*  <value>
+                // t1 = *  HWINTRINSIC simd16 double CreateScalarUnsafe
+                //      *  STORE_LCL_VAR simd16<Vector128`1>
+                // t2 =    LCL_VAR   simd16<Vector128`1>
+                // t3 =    LCL_VAR   simd16<Vector128`1>
+                //      /--*  t2     simd16
+                //      +--*  t3     simd16
+                // t0 = *  HWINTRINSIC simd16 float MoveLowToHigh
+
+                node->gtOp1 = op1;
+
+                LIR::Use op1Use(BlockRange(), &node->gtOp1, node);
+                ReplaceWithLclVar(op1Use);
+
+                node->gtOp2 = comp->gtClone(node->gtOp1);
+                BlockRange().InsertBefore(node, node->gtOp2);
+
+                node->gtHWIntrinsicId = NI_SSE_MoveLowToHigh;
+                node->gtSIMDBaseType  = TYP_FLOAT;
+
+                break;
+            }
+
+            default:
+            {
+                unreached();
+            }
+        }
+
+        return;
+    }
+
+    // We have the following:
+    //
+    // /--*  <valueFirst>
+    // +--*  <value...>
+    // +--*  <valueLast>
+    // *  HWINTRINSIC simd16 T Create
+
+    if (intrinsicId == NI_Vector256_Create)
+    {
+        assert(comp->compIsaSupportedDebugOnly(InstructionSet_AVX));
+
+        // We will be constructing the following parts:
+        //
+        // t2 =    CNS_INT   int    0
+        //      /--*  t?     simd32
+        //      +--*  t?     simd16
+        //      +--*  t?     int
+        // t0 = *  HWINTRINSIC simd32 T InsertVector128
+
+        int halfArgCnt = argCnt / 2;
+        assert((halfArgCnt * 2) == argCnt);
+
+        argList = op1->AsArgList();
+
+        for (int i = 0; i < halfArgCnt; i++)
+        {
+            op2     = argList;
+            argList = argList->Rest();
+        }
+
+        op2->AsArgList()->gtOp2 = nullptr;
+        op2                     = argList;
+
+        if (halfArgCnt == 2)
+        {
+            argList = op1->AsArgList();
+
+            tmp = argList->Rest()->Current();
+            op1 =
+                comp->gtNewSimdHWIntrinsicNode(TYP_SIMD16, argList->Current(), tmp, NI_Vector128_Create, baseType, 16);
+            BlockRange().InsertAfter(tmp, op1);
+            LowerNode(op1);
+
+            argList = op2->AsArgList();
+
+            tmp = argList->Rest()->Current();
+            op2 =
+                comp->gtNewSimdHWIntrinsicNode(TYP_SIMD16, argList->Current(), tmp, NI_Vector128_Create, baseType, 16);
+            BlockRange().InsertAfter(tmp, op2);
+            LowerNode(op2);
+        }
+        else
+        {
+            tmp = op2->AsArgList()->Current();
```

Comment:
I'm quite confused by this, and I suspect I'm missing something. You've handle the 4-argument case above, and this handles 2. Where do you handle 8 arguments?

Reply:
This one actually handles everything that is `> 4` and I can add an assert + comment to clarify.

For `Vector256` (which this code path is restricted to) we will have 4, 8, 16, or 32 operands, so in all cases we have a `GT_LIST`.
Both paths construct the upper and lower `Vector128` portions and then combine them. The for loop [here](https://github.com/dotnet/runtime/pull/35857/files#diff-55de8576070c2b77ed274e8e03c6676fR1639) ensures that op1 points to the first operand and op2 points to the 3rd, 5th, 9th, or 17th operand.

The first path needs to exist since it will be 2 operands per half, and se we need to track them in `gtOp1`/`gtOp2` rather than in a `GT_LIST`
While the second path can just use the original list, split into two halves with the 2nd, 4th, 8th, or 16th operand no longer have a `gtOp2` (which normally points to the next operand in the list).

Reply:
Ah yes, I see now that `op1` and `op2` are still lists, and the two `NI_Vector128_Create` intrinsics are then recursively lowered. I might ask that you create a new `GenTree*` variable for those rather than reusing `op1`. I know that the JIT reuses variables all over the place, but I don't think it's a great practice.

Reply:
Added a comment explaining the split and what values are expected where, using new variables named `lo` and `hi` to track the lower and upper halves of the Vector256 being created.